dashboard/app/api.go

```diff
@@ -1551,31 +1552,52 @@ func needReproForBug(c context.Context, bug *Bug) bool {
 	return timeSince(c, bug.LastReproTime) >= reproStalePeriod
 }
 
-func putText(c context.Context, ns, tag string, data []byte, dedup bool) (int64, error) {
+var dedupText = &struct{}{}   // attempt to deduplicate the content
+var cutInMiddle = &struct{}{} // when the text has to be truncated, cut the middle part
```

Comment:
Is it really just exact middle bytes?
Why are middle bytes useful for logs? I would assume tail of the log is useful...

Reply:
You suggest always cutting the suffix for all types of tags?

Reply:
I am not sure what's in the logs that don't fit into 1MB compressed.
For other text types it's not expected that they will be over 1MB compressed, e.g. for a kernel config there is really no right cut if we have to cut it.
Generally, if we have panic_on_warn, then the oops message is at the very end, if we keep middle, then we can discard the actual oops message. Maybe we should we keep the tail always.

Reply:
The truncated log there does not have any of the kernel output. The last executed programs are also closer to the end of the log. So for that example it looks more reasonable to keep the tail. I assume keeping middle will also cut the oops message.

Reply:
FWIW it was not about keeping the middle, but rather about cutting the middle, so the panic would have remained.

But in any case, it does really look like we can just always iterate over the suffixes. I updated the PR.